### PR TITLE
Dynamic input and cfg file creation for vNIC interface.[ Do not merge ]

### PR DESCRIPTION
**Pull request by FarooqAbdulla02**
This code supports to dynamically create "configuration file" and
"input file" for virtual interface with few required parameters.

### CODE REVIEW

**Comment by FarooqAbdulla02** [comment]
A. show info for virtual interface :
# ./pci_info.py --vnic env3 --show-info
[{'adapter_type': 'vnic',
'driver': 'ibmvnic',
'interfaces': ['env3', 'env4', 'env5'],
'mac_id': '02af7e50c203',
'macaddress': '02:af:7e:50:c2:03',
'public_interface_ip': '9.40.192.184'}]

B. create config without additional params:
# ./pci_info.py --vnic env3 --create-config
19:39:41 INFO : python avocado-setup.py --run-suite host_io_env3_vnic_dymamic_fvt --input-file io_input.txt
# cat io_input.txt
[io_env3_vnic_dymamic_fvt]
host_interfaces = "env3 env4"
bond_interfaces = "env3 env4"
htx_host_interfaces = "env3 env4"
host_public_ip = "9.40.192.184"
interface = "env3"
module = "ibmvnic"
macaddress = "02:af:7e:50:c2:03"
mac_id = "02af7e50c203"

C. create config with additional params
#./pci_info.py --vnic env3 --additional-params device_ip=192.168.100.184,peer_ip=192.168.100.104,netmask=255.255.255.0,
host_ip=192.168.100.184,netmasks=255.255.255.0,peer_ips="192.168.100.104 192.168.110.104",peer_interfaces="env4 env5" --create-config

  19:35:11 INFO    : python avocado-setup.py --run-suite host_io_env3_vnic_dymamic_fvt --input-file io_input.txt
    # cat io_input.txt
    [io_env3_vnic_dymamic_fvt]
    host_interfaces = "env3 env4"
    bond_interfaces = "env3 env4"
    htx_host_interfaces = "env3 env4"
    host_public_ip = "9.40.192.184"
    interface = "env3"
    module = "ibmvnic"
    macaddress = "02:af:7e:50:c2:03"
    mac_id = "02af7e50c203"
    device_ip = 192.168.100.184
    peer_ip = 192.168.100.104
    host_ip = 192.168.100.184
    netmask = 255.255.255.0
    netmasks = 255.255.255.0
    peer_ips = 192.168.100.104 192.168.110.104
    host_ips = "110.10.10.212 111.10.10.212"
    peer_interfaces = env4 env5

File: config/wrapper/pci_input.conf [hunk 1]
```diff
@@ -98,6 +98,17 @@ host_interfaces = interfaces:all
 pci_device,functions:0
 module = driver
 
+
+[io_vnic_dynamic_fvt]
```
Comment (abdhaleegit) [inline config/wrapper/pci_input.conf:102]: just keep this name as io_vnic_fvt

File: lib/virtual.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+    :return: string of MAC address.
+    '''
+    try:
+        for line in runcmd("ip a s dev %s" % interface)[1].splitlines():
+            if 'link/ether' in line:
+                mac_address = line.split()[1]
+                return mac_address
+    except Exception as e:
+        print(f'Interface not found {e}')
```
Comment (abdhaleegit) [inline lib/virtual.py:39]: instead print raise the error

File: pci_info.py [hunk 3]
```diff
@@ -34,107 +35,154 @@
 input_path = "io_input.txt"
 INPUTFILE = configparser.ConfigParser()
 INPUTFILE.optionxform = str
-
 logger = logger_init(filepath=BASE_PATH).getlogger()
 
 
-def create_config(pci_list):
+def create_config_inputs(orig_cfg, new_cfg, inputfile, interface, config_type):
```
Comment (abdhaleegit) [inline pci_info.py:41]: you do not need to pass old and and new cfg file name.. instead it should fetch default cfg files in fvt wrapper and append the interface name like io_vnic_env3_fvt.cfg to the file also.. to input file io-input.txt .. not sure why you are passing them as parameter

**Review by abdhaleegit — CHANGES_REQUESTED** [review]
does it not affect the old pci network bucket... looks like a little change in the create_config flow

**Comment by abdhaleegit** [comment]
@FarooqAbdulla02 

As discussed, couple of suggestions:

1. Include also the constant parameters values into the final input file, also fetch is_mlx_driver, default tx/rx etc and ommit slot/sriovadapter/port params as vnic is already created
2. Add support to create input file with out vnic interface specified i.e just — vnic should collect available vnic interface data into input.. this is must for end to end flow
3. Add a new bucket io_vnic_stress_fvt.cfg containing only runnable tests with dynamic inputfile .. remove dependency of hmc/vios (device add/delete no in scope)
4. Follow the old naming conventions for bucket name I.e [io_vnic_env3_fvt] .. as per your code it follows this [io_env3_vnic_dymamic_fvt]
5. Make sure the it creates input/cfg for mix of vnic and sriov in single lpar
6. finally rebase the code .. it shows merge conflict..  

Note: make sure the API's works for both RHEL and SLES O

**Comment by abdhaleegit** [comment]
close this one once the new PR is merged

**Comment by FarooqAbdulla02** [comment]
same issue has been tracked on https://github.com/lop-devops/tests/pull/341 and closing this as duplicate.

**Pull request closed without merging**